### PR TITLE
Add new Tracks around WooPay Save My Info checkbox

### PR DESCRIPTION
**Pull request by malithsen**
#### Changes proposed in this Pull Request

This PR improves Tracks coverage around WooPay Save My Info usage. 
- `woocommerceanalytics_woopay_offered` event was renamed to `wcpay_checkout_woopay_save_my_info_offered` to remove the ambiguity around the source.
- Following two events are added
  - `wcpay_checkout_woopay_save_my_info_country_click`
  - `wcpay_checkout_woopay_save_my_info_mobile_enter`

#### Testing instructions

* As a shopper, add a product to the cart.
* Go to the classic checkout page.
* Enter an email that's not already registered with WooPay.
* Wait for the "Save My info" checkbox to appear
* In ~5 minutes, check Tracks Live view for the `wcpay_checkout_woopay_save_my_info_offered` event
* Enter a valid phone number in the input box under the "Save My info" section.
* In ~5 minutes, check Tracks Live view for the `checkout_woopay_save_my_info_mobile_enter` event
* Click the dropdown menu to change the country code.
* In ~5 minutes, check Tracks Live view for the `wcpay_checkout_woopay_save_my_info_country_click` event

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7973` or branch name `add/save-my-info-tracks` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/save-my-info-tracks) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8985ca0e1dad7fbe9c1e8213fcd87e61f60d3e3e
- Build time: 2024-01-12 20:37:48 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/tracks/index.js [hunk 1]
```diff
@@ -108,12 +108,16 @@ const events = {
...
-	WOOPAY_OFFERED: 'woopay_offered',
+	WOOPAY_OFFERED: 'checkout_woopay_save_my_info_offered',
 	WOOPAY_AUTO_REDIRECT: 'checkout_woopay_auto_redirect',
 	WOOPAY_SKIPPED: 'woopay_skipped',
 	WOOPAY_BUTTON_LOAD: 'woopay_button_load',
 	WOOPAY_BUTTON_CLICK: 'woopay_button_click',
+	WOOPAY_SAVE_MY_INFO_COUNTRY_CLICK:
+		'checkout_woopay_save_my_info_country_click',
 	WOOPAY_SAVE_MY_INFO_CLICK: 'checkout_save_my_info_click',
+	WOOPAY_SAVE_MY_INFO_MOBILE_ENTER:
```
Comment (hsingyuc) [inline client/tracks/index.js:119]: Do we track it somewhere?

**Review by hsingyuc — COMMENTED** [review]
@malithsen Thanks for working on this! I left a question below.

File: client/tracks/index.js (hunk 1 above)
Comment (malithsen) [reply client/tracks/index.js:119]: Oof thanks for catching that. Added in https://github.com/Automattic/woocommerce-payments/pull/7973/commits/dce43950dd98d2c42c331dfd074e098001dbeb7c

**Review by hsingyuc — APPROVED** [review]
@malithsen Tests are as expected and LGTM!

**Pull request merged**